### DISCUSSION TITLE
Bubble self-energy calculation with LO-TO splitting

### DISCUSSION THREAD

**Discussion by eargnawk** [ttadano/alamode#65] [Q&A] [answered]
Dear Dr. Tadano,

Thanks for your awesome code development.
I recently tried to calculate bubble self-energy.
I was successful to create effective 2nd order force constants using `dfc2` script for calculating bubble self-energy with `RTA` mode at each temperature.
However, when I see the data in `result.self*`, the files are looked as follows:
```
# xk =               0              0              0
# mode = 13
## T[K], Freq (cm^-1), omega (cm^-1), Self.real (cm^-1), Self.imag (cm^-1)
       800        512.289         0        29.7724   -1.39377e-32
       800        512.289       0.1        29.7725    5.20816e-05
       800        512.289       0.2        29.7729    0.000181648
       800        512.289       0.3        29.7734    0.000386356
       800        512.289       0.4        29.7742    0.000693036
...
```
```
# xk =               0              0              0
# mode = 14
## T[K], Freq (cm^-1), omega (cm^-1), Self.real (cm^-1), Self.imag (cm^-1)
       800        512.289         0        29.7769    4.38787e-33
       800        512.289       0.1        29.7771    6.58749e-05
       800        512.289       0.2        29.7774    0.000226099
       800        512.289       0.3         29.778    0.000466992
       800        512.289       0.4        29.7787    0.000807992
...
```
```
# xk =               0              0              0
# mode = 15
## T[K], Freq (cm^-1), omega (cm^-1), Self.real (cm^-1), Self.imag (cm^-1)
       800        512.289         0        29.7758   -3.18346e-34
       800        512.289       0.1         29.776    6.89256e-05
       800        512.289       0.2        29.7763    0.000236417
       800        512.289       0.3        29.7769    0.000486661
       800        512.289       0.4        29.7777    0.000838017
...
```

Even though I included `NONANALYTIC=2` or `NONANALYTIC=3` tag with `*.born` file, it seems that the frequency values in the second-column are the phonon frequencies without consideration of LO-TO splitting.
Is it okay just using self.real values for bubble self-energy consideration of each mode even if the frequencies of LO and TO are different in the SCPH band structure?

Thanks.

**Comment by eargnawk** [comment, marked as answer]
Now I just realized there is a difference between q -> 0 and q = 0. Sorry for the confusion. Thanks.

**Reply by ttadano** [reply]
Yes, as you have just realized, the non-analytic term is divergent when q is exactly 0. So the code does not compute the nonanalytic correction at q=0. Perhaps, one may approximately calculate the NA term at q=0 by taking the average of the NA terms at q=(0+, 0, 0), q=(0,0+,0), q=(0,0,0+) with 0+ being an infinitesimal value, but I'm not sure if this procedure generally works.
